js/vector.js

```diff
@@ -0,0 +1,63 @@
+class Vector {
+
+    constructor(x, y) {
+        if (arguments.length === 0) {
+            this.x = 0;
+            this.y = 0;
+        } else {
+            this.x = x;
+            this.y = y;
+        }
+    }
+
+    add(v, y) {
+        if (arguments.length === 1) {
+            this.x += v.x;
+            this.y += v.y;
+        } else {
+
+            this.x = this.x + v;
+            this.y = this.y + y;
+        }
+    }
+
+    subtract(v, y) {
+        if (arguments.length === 1) {
+            this.x -= v.x;
+            this.y -= v.y;
+        } else {
+            this.x -= v;
+            this.y -= y;
+        }
+    }
+
+    multiply(s) {
```

Comment:
let's rename s to "factor" or something like that. That makes it clear that it's a number and not another vector